pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
I think of nodes as being very dynamic, so I think it would have to be Node state being "up" rather than the cluster state.  But pedantry aside, I'd rather avoid having a complex Node lifecycle if we can avoid it - it's been the cause of many bugs in my own code dealing with cloud APIs when I assumed that property X would be set but it was actually set asynchronously with an unknown delay.

I do think that we will end up exposing all that extra information in the end (I'm looking in particular at NodeSystemInfo).  I did originally try an alternate implementation where I added a typed Zone data structure to the Node, but I actually think labels are "more kubernetes".  Having labels lets us use the selector approach (which I think is a great kubernetes 'core idea') - i.e. we get to use the NodeSelector predicates in the scheduler.

The other downside of having a strongly typed Zone object is that we want to add this information to every type that has zone restrictions; Nodes & Volumes for a simple multi-AZ kubernetes, probably every type in full federated kubernetes (Ubernetes).

In any case, we can start with labels and map labels <-> objects later if we do decide that's what we want to do; as these are in the "kubernetes.io" reserved prefix.

This is part of the Ubernetes (Lite) effort, so while today "the cluster can function without them", that won't be true for some of the cool stuff that is coming :-)  You can check out the branch where I have multi-AZ working if you want to see where this is going: https://github.com/justinsb/kubernetes/tree/multizone_with_labels

A potential compromise: we could gate this behind an experimental flag if you'd like?  I'd prefer not to, but it wouldn't really impair the user experience if we automatically set it for the user when they are running multi-AZ.